integration/inspektor-gadget/integration_test.go

```diff
@@ -1325,49 +1314,31 @@ func TestTcptracer(t *testing.T) {
 		Cmd:          fmt.Sprintf("$KUBECTL_GADGET trace tcp -n %s -o json", ns),
 		StartAndStop: true,
 		ExpectedOutputFn: func(output string) error {
-			expectedEntries := []*tcpTypes.Event{
-				{
-					Event:     BuildBaseEvent(ns),
-					Comm:      "wget",
-					IPVersion: 4,
-					Daddr:     "1.1.1.1",
-					Dport:     80,
-					Operation: "connect",
-				},
-				{
-					Event:     BuildBaseEvent(ns),
-					Comm:      "wget",
-					IPVersion: 4,
-					Daddr:     "1.1.1.1",
-					Dport:     80,
-					Operation: "close",
-				},
-				{
-					Event:     BuildBaseEvent(ns),
-					Comm:      "wget",
-					IPVersion: 4,
-					Daddr:     "1.1.1.1",
-					Dport:     443,
-					Operation: "connect",
-				},
+			expectedEntry := &tcpTypes.Event{
+				Event:     BuildBaseEvent(ns),
+				Comm:      "wget",
+				IPVersion: 4,
+				Dport:     443,
+				Operation: "connect",
 			}
 
 			normalize := func(e *tcpTypes.Event) {
 				e.Node = ""
 				e.Pid = 0
 				e.Saddr = ""
+				e.Daddr = ""
```

Comment:
I suppose we have to make this because we're accessing the API server by using a URL. I feel the idea of testing with a server (nginx) or similar will allow us to perform a deeper test without relying on DNS.

Reply:
True. Now that I am thinking how to propagate this address/port, deploying own pod sounds simpler.

Reply:
I implemented a function to get the IP of a pod in https://github.com/inspektor-gadget/inspektor-gadget/commit/d012b5cfabb99370a0ff0b737c9361815dea643d, this could help in this case too. (probably has to be changed a bit to allow different pod names but that's the idea).